tests/rbd/rbd_tier0.py

```diff
@@ -0,0 +1,317 @@
+        )
+        log.info(f"Updated date : {update_date}")
+        if update_date.strip() != date:
+            log.error("Setting image metadata last_update failed")
+            return 1
+
+        rbd.image_meta(
+            action="remove",
+            image_spec=image_spec,
+            key="last_update",
+        )
+        update_date = rbd.image_meta(
+            action="get", image_spec=image_spec, key="last_update"
+        )
+        log.info(f"Updated date : {update_date}")
+        if update_date != 1:
+            log.error("Removing image metadata last_update failed")
+            return 1
+
+        # Create snapshot, protect, clone, flatten, resize, remove, rollback
+        if rbd.snap_create(pool, image, snap):
+            log.error(f"Snapshot create failed for {snap_name}")
+            return 1
+
+        if rbd.protect_snapshot(snap_name):
+            log.error(f"Snapshot protect failed for {snap_name}")
+            return 1
+
+        if rbd.create_clone(snap_name, pool, clone):
+            log.error(f"Clone create failed for {clone}")
+            return 1
+
+        if rbd.flatten_clone(pool, clone):
+            log.error(f"Clone flatten failed for {clone}")
+            return 1
+
+        resize_int = size_int + 2
+        resize = f"{resize_int}{size.replace(str(size_int),'')}"
+        if rbd.image_resize(pool, clone, resize):
+            log.error(f"Clone resize failed for {clone}")
+            return 1
+
+        rbd_info = rbd.image_info(pool_name=pool, image_name=clone)
+        if int(rbd_info["size"] / (1024 * 1024 * 1024)) != resize_int:
+            log.error(f"Image resize failed on clone {clone}")
+            return 1
+
+        # verify rollback functionality, resize original image, rollback to the snap
+        # verify that original image size is preserved and resize is reversed
+        # the steps given below perform this verification
+
+        if rbd.image_resize(pool, image, resize):
+            log.error(f"Image resize failed for {image}")
+            return 1
+
+        rbd_info = rbd.image_info(pool_name=pool, image_name=image)
+        if int(rbd_info["size"] / (1024 * 1024 * 1024)) != resize_int:
+            log.error(f"Image resize failed on {image}")
+            return 1
+
+        if rbd.snap_rollback(snap_name):
+            log.error(f"Snap roll back command failed for {snap_name}")
+
+        rbd_info = rbd.image_info(pool_name=pool, image_name=image)
+        if rbd_info["size"] / (1024 * 1024 * 1024) != size_int:
+            log.error(f"Snap roll back failed on {image}")
+            return 1
+
+        # Add verification for snap rollback
+        if rbd.unprotect_snapshot(snap_name):
+            log.error(f"Snapshot unprotect failed for {snap_name}")
+            return 1
+
+        if rbd.snap_remove(pool, image, snap):
+            log.error(f"Snapshot remove failed for {snap_name}")
+            return 1
+
+        # Rename (move) to a different pool and verify
```

Comment:
Actually we are not moving it to a different pool right ?

Reply:
changed the statement according to the action performed.